modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractCSharpCodegen.java

```diff
@@ -454,6 +461,19 @@ private void postProcessEnumRefs(final Map<String, Object> models) {
                         var.isPrimitiveType = true;
                     }
                 }
+                for (CodegenProperty var : model.vars) {
```

Comment:
Question:
why is this line (some lines above) not covering all `enums`
```
for (CodegenProperty var : model.allVars) 
```

Reply:
It should. That could be a bug with allVars.

I'll look into it in the coming week.